services/ymax-planner/src/gas-estimation.ts

```diff
@@ -0,0 +1,81 @@
+import type { AxelarChain } from '@agoric/portfolio-api/src/constants';
+import { gasLimitEstimates } from './support.ts';
+
+const AGORIC_CHAIN = 'agoric';
+const BLD_TOKEN = 'ubld';
+
+export const prepareGasEstimator =
+  ({
+    axelarApiAddress,
+  }: {
+    axelarApiAddress: string;
+  }) =>
+  (chainName: AxelarChain) => {
+    const axelarEstimateGasUrl = `${axelarApiAddress}/gmp/estimateGasFee`;
+
+    const queryAxelarGasAPI = async (
+      sourceChain: AxelarChain | 'agoric',
+      destinationChain: AxelarChain | 'agoric',
+      gasLimit: bigint,
+      gasToken?: string,
+    ) => {
+      const response = await fetch(axelarEstimateGasUrl, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          sourceChain,
+          destinationChain,
+          gasLimit: gasLimit.toString(),
+          sourceTokenSymbol: gasToken,
+          gasMultiplier: '1',
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const data = Number(await response.json());
+      return data;
+    };
+
+    const getWalletEstimate = async () => {
+      const gasLimit = gasLimitEstimates['WALLET'];
+
+      const estimate = await queryAxelarGasAPI(
+        AGORIC_CHAIN,
+        chainName,
+        gasLimit,
+        BLD_TOKEN,
+      );
+
+      return BigInt(estimate);
+    };
+
+    const getFactoryContractEstimate = async () => {
+      const gasLimit = gasLimitEstimates['AAVE_SUPPLY'];
+
+      const estimate = await queryAxelarGasAPI(
+        AGORIC_CHAIN,
+        chainName,
+        gasLimit,
+        BLD_TOKEN,
+      );
+
+      return BigInt(estimate);
+    };
+
+    const getReturnFeeEstimate = async () => {
+      const fee = await queryAxelarGasAPI(chainName, AGORIC_CHAIN, 1n);
+
+      return BigInt(fee);
+    };
```

Comment:
These get simpler if you take my above suggestion to return bigint from `queryAxelarGasAPI`:
```suggestion
    const getWalletEstimate = () =>
      queryAxelarGasAPI(
        AGORIC_CHAIN,
        chainName,
        gasLimitEstimates['WALLET'],
        BLD_TOKEN,
      );

    const getFactoryContractEstimate = () =>
      queryAxelarGasAPI(
        AGORIC_CHAIN,
        chainName,
        gasLimitEstimates['AAVE_SUPPLY'],
        BLD_TOKEN,
      );

    const getReturnFeeEstimate = () =>
      queryAxelarGasAPI(chainName, AGORIC_CHAIN, 1n);
```